app/javascript/dashboard/routes/dashboard/helpcenter/components/ArticleSearch/SearchPopover.vue

```diff
@@ -0,0 +1,165 @@
+    ArticleView,
+  },
+  mixins: [clickaway, portalMixin],
+  props: {
+    selectedPortalSlug: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      searchQuery: '',
+      isLoading: false,
+      searchResults: [],
+      activeId: '',
+      debounceSearch: () => {},
+    };
+  },
+  computed: {
+    articleViewerUrl() {
+      const article = this.activeArticle(this.activeId);
+      if (!article) return undefined;
+      const isDark = document.body.classList.contains('dark');
+
+      const url = new URL(article.url);
+      url.searchParams.set('show_plain_layout', 'true');
+
+      if (isDark) {
+        url.searchParams.set('theme', 'dark');
+      }
+
+      return `${url}`;
+    },
+    searchResultsWithUrl() {
+      return this.searchResults.map(article => ({
+        ...article,
+        localeName: this.localeName(article.category.locale || 'en'),
+        url: this.generateArticleUrl(article),
+      }));
+    },
+  },
+  mounted() {
+    // Load popular articles on mount
+    this.fetchArticlesByQuery(this.searchQuery);
+    this.debounceSearch = debounce(this.fetchArticlesByQuery, 500, true);
+    document.body.addEventListener('keydown', this.closeOnEsc);
+  },
+  beforeDestroy() {
+    document.body.removeEventListener('keydown', this.closeOnEsc);
+  },
+  methods: {
+    generateArticleUrl(article) {
+      const host =
+        window.chatwootConfig.helpCenterURL || window.chatwootConfig.hostURL;
+      return `${host}/hc/${this.selectedPortalSlug}/articles/${article.slug}`;
+    },
+    activeArticle(id) {
+      return this.searchResultsWithUrl.find(article => article.id === id);
+    },
+    onSearch(query) {
+      this.searchQuery = query;
+      this.activeId = '';
+      this.debounceSearch(query);
+    },
+    onClose() {
+      this.$emit('close');
+      this.searchQuery = '';
+      this.activeId = '';
+      this.searchResults = [];
+    },
+    async fetchArticlesByQuery(query) {
+      try {
+        const sort = query ? undefined : 'views';
+        this.isLoading = true;
+        this.searchResults = [];
+        const { data } = await ArticlesAPI.searchArticles({
+          portalSlug: this.selectedPortalSlug,
+          query,
+          sort,
+        });
+        this.searchResults = data.payload;
+        this.isLoading = true;
+      } catch (error) {
+        // Show something wrong message
+      } finally {
+        this.isLoading = false;
+      }
+    },
+    handlePreview(id) {
+      this.activeId = id;
+    },
+    onBack() {
+      this.activeId = '';
+    },
+    onInsert(id) {
+      const article = this.activeArticle(id || this.activeId);
+
+      this.$emit('insert', article);
+      bus.$emit(
+        'newToastMessage',
+        this.$t('HELP_CENTER.ARTICLE_SEARCH.SUCCESS_ARTICLE_INSERTED')
+      );
+    },
+    closeOnEsc(e) {
+      if (isEscape(e) && !isActiveElementTypeable(e)) {
+        e.preventDefault();
+        this.onClose();
+      }
```

Comment:
Please use `buildHotKeys` method.

Reply:
We just need this to detect `Escape`, is `buildHotKeys` necessary for this usecase?